### PR TITLE
`key_share`: add HD-related methods to `DirtyKeyInfo`

### PR DESCRIPTION
**Pull request by survived**

### CODE REVIEW

File: key-share/src/lib.rs [hunk 1]
```diff
@@ -441,6 +441,33 @@ impl<E: Curve> DirtyCoreKeyShare<E> {
     }
 }
 
+#[cfg(feature = "hd-wallets")]
+impl<E: Curve> DirtyCoreKeyShare<E> {
+    /// Checks whether the key is HD-capable
+    pub fn is_hd_wallet(&self) -> bool {
+        (**self).is_hd_wallet()
+    }
```
Comment (maurges) [inline key-share/src/lib.rs:449]: Aren't these methods available automatically with DirtyCoreKeyShare implementing Deref into DirtyKeyInfo?

Comment (survived) [reply key-share/src/lib.rs:449]: They should be, but it’s still a breaking change to remove a method in public API (even if it’s still available via derefing)

Comment (survived) [reply key-share/src/lib.rs:449]: E.g. using DirtyCoreKeyShare::is_hd_wallet compiles before the removal, but doesn’t compile after the removal, derefing doesn’t work here

Comment (maurges) [reply key-share/src/lib.rs:449]: Since when do you care about breaking the API? Just bump the version :>

**Review by maurges — APPROVED** [review]

File: key-share/src/lib.rs (hunk 1 above)
Comment (survived) [reply key-share/src/lib.rs:449]: Since always! I never was a fan of updating all the upstream deps, I only do that when there's no choice

**Pull request merged**